source/appModules/teamtalk5classic.py

```diff
@@ -0,0 +1,245 @@
+# appModules/teamtalk5classic.py
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2012-2019 NV Access Limited, Doug Lee, Tyler Spivey, Bill Dengler
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+
+import api
+import appModuleHandler
+import controlTypes
+import scriptHandler
+import speech
+import ui
+
+from NVDAObjects.behaviors import ProgressBar
+from NVDAObjects.IAccessible import IAccessible
+
+# Constants
+# Translators: Status text in the TeamTalk voice conferencing client.
+LBL_CONNECTED = _("Connected")
+# Translators: Status text in the TeamTalk voice conferencing client.
+LBL_PTT = _("Push To Talk")
+# Translators: Status text in the TeamTalk voice conferencing client.
+LBL_VOX = _("Voice Activation")
+# Translators: Status text in the TeamTalk voice conferencing client.
+LBL_VIDEO = _("Video")
+# Translators: Status text in the TeamTalk voice conferencing client.
+# Note: this string is slightly different from the actual toolbar text
+# to avoid a translatable string conflict.
+LBL_DESKTOP = pgettext("TeamTalk", "Desktop")
+# Translators: Status text in the TeamTalk voice conferencing client.
+LBL_STREAM = _("Stream To Channel")
+# Translators: Status text in the TeamTalk voice conferencing client.
+LBL_MUTE = _("Mute All")
+# Translators: Status text in the TeamTalk voice conferencing client.
+LBL_RECORD = _("Store Audio")
+# Translators: Status text in the TeamTalk voice conferencing client.
+LBL_MSG = _("Show Channel Messages")
+
+#: The names, in order, of the toolbar items.
+#: The count must match exactly or this module will not function.
+#: The name "sep" is a placeholder for separators in the toolbar.
+#: These are name sets per client version.
+#: Stored in reverse so current clients find the right list faster.
+_msgNames = {
+	"4.5": [
+		LBL_CONNECTED,
+		"sep",
+		LBL_PTT,
+		LBL_VOX,
+		LBL_VIDEO,
+		LBL_DESKTOP,
+		LBL_STREAM,
+		"sep",
+		LBL_MUTE,
+		LBL_RECORD,
+		"sep",
+		LBL_MSG,
+	],
+	"4.3": [
+		LBL_CONNECTED,
+		"sep",
+		LBL_PTT,
+		LBL_VOX,
+		LBL_VIDEO,
+		LBL_DESKTOP,
+		LBL_MUTE,
+		LBL_RECORD,
+		"sep",
+		LBL_MSG,
+	],
+	"4.2": [
+		LBL_CONNECTED,
+		"sep",
+		LBL_PTT,
+		LBL_VOX,
+		"sep",
+		LBL_MUTE,
+		LBL_RECORD,
+		"sep",
+		LBL_MSG,
+	],
+	"4.1": [
+		LBL_CONNECTED,
+		"sep",
+		LBL_PTT,
+		LBL_VOX,
+		"sep",
+		LBL_MUTE,
+		"sep",
+		LBL_MSG,
+	],
+}
+
+# window control IDs
+ID_CHATOUT = 1204
+ID_CHATIN = 1206
+ID_BANNEDLIST = 1326
+ID_UNBANNEDLIST = 1327
+ID_TOOLBAR = 59392
+ID_VU = 1004
+
+
+class TTToolBarInfo(object):
+	"""Info about one TeamTalk toolbar and its items.
+	"""
+
+	def __init__(self, NVDAObj=None):
+		if not NVDAObj:
+			# Use the IAccessible for the active TeamTalk instance's toolbar.
+			NVDAObj = self.getToolBar()
+			if not NVDAObj:
+				# Translators: Reported when status information is unavailable
+				# in the TeamTalk voice conferencing client.
+				ui.message(_("Status unavailable"))
```

Comment:
When this object is initiated and there is no object, status unavailable is spoken, but creating the object still succeeds. I'd rather have initialisation of the object raise an error, catch that and use the ui.message there instead.

Reply:
> When this object is initiated and there is no object, status unavailable is spoken, but creating the object still succeeds. I'd rather have initialisation of the object raise an error, catch that and use the ui.message there instead.

This was from the original add-on, presumably to allow `script_sayToolBarInfo` to be implemented. Can you think of another way to do that now that I've moved the old code over?